server/src/main/java/org/elasticsearch/action/ActionListener.java

```diff
@@ -425,6 +426,16 @@ public boolean equals(Object obj) {
         }
     }
 
+    /**
+     * @return A listener which (if assertions are enabled) wraps around the given delegate and asserts that it is called at least once.
+     */
+    static <Response> ActionListener<Response> assertAtLeastOnce(ActionListener<Response> delegate) {
+        if (Assertions.ENABLED) {
+            return ActionListener.runBefore(delegate, LeakTracker.INSTANCE.track(delegate)::close);
```

Comment:
This actually ends up being `assertExactlyOnce` with this implementation because `runBefore` invokes `assertOnce`. I wonder if we should call the method `assertExactlyOnce`.

Reply:
Ah good point, well spotted. Let's fix that so that this method returns a listener can be called at least once (but still using `LeakTracker.INSTANCE.track(delegate)`).